pkg/server/cluster_definition_handler.go

```diff
@@ -0,0 +1,95 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+)
+
+const authTokenRequestURL = "https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/token"
+const pullSecretRequestURL = "https://api.openshift.com/api/accounts_mgmt/v1/access_token"
+
+type ClusterDefinition struct {
+	ClusterName string `json:"clusterName"`
+	DNSDomain   string `json:"DNSDomain"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+	PullSecret  string `json:"pullSecret"`
+}
+
+type DeveloperTokenResponse struct {
+	ExpiresIn        int    `json:"expires_in"`
+	RefreshExpiresIn int    `json:"refresh_expires_in"`
+	RefreshToken     string `json:"refresh_token"`
+	TokenType        string `json:"token_type"`
+	NotBeforePolicy  int    `json:"not-before-policy"`
+	SessionState     string `json:"session_state"`
+	AccessToken      string `json:"access_token"`
+}
+
+func fetchAuthToken(Username string, Password string) (tokenData *DeveloperTokenResponse, err error) {
+	resp, err := http.PostForm(authTokenRequestURL, url.Values{
+		"grant_type": {"password"},
+		"client_id":  {"uhc"},
+		"username":   {Username},
+		"password":   {Password},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&tokenData)
+	if err != nil {
+		return nil, err
+	}
+
+	return tokenData, nil
+}
+
+func fetchPullSecret(tokenData *DeveloperTokenResponse) (pullSecret *map[string]interface{}, err error) {
+	req, err := http.NewRequest("POST", pullSecretRequestURL, nil)
+	req.Header.Add("accept", "application/json")
+	req.Header.Add("authorization", "bearer "+tokenData.AccessToken)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&pullSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return pullSecret, nil
+}
+
+func CreateClusterDefinition(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think the handler should be in `handlers.go`, and the other code should be in `pkg/integration/token` (or similar). That way the CLI can also get the pull secret.

Reply:
Done